pkg/ssh/ssh.go

```diff
@@ -27,15 +28,19 @@ func Dial(options *ConnectionDialOptions, kind EngineMode) (*ssh.Client, error)
 }
 
 func Exec(options *ConnectionExecOptions, kind EngineMode) (string, error) {
+     return ExecWithInput(options, kind, nil)
+}
+
+func ExecWithInput(options *ConnectionExecOptions, kind EngineMode, input *io.Reader) (string, error) {
```

Comment:
```suggestion
func ExecWithInput(options *ConnectionExecOptions, kind EngineMode, input io.Reader) (string, error) {
```
io.Reader is already an interface so this should not be a pointer to the interface as it adds extra indirection for no reason.